Update footer to latest ds-toolbox utility classes

ashley-hebler opened the pull request:
#### What's this PR do?
Syncs up some updates @AndrewGibson27 and I have been making in our CSS framework for UMP and otherwise with the current site footer.

#### Why are we doing this? How does it help us?
This is just another iteration of our overall CSS refactor efforts

#### How should this be manually tested?
```
make backing
make
yarn run dev
make restart
```
Make sure the footer doesn't look bonkers

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
Adds 3kb of CSS, so not the end of the world, but once we utilize more of the helper classes in other places, we can probably cut down the repo specific CSS to counter this increase.

#### What are the relevant tickets?
Nothing specific, but improvements came as a byproduct of the UMP portal work

#### Have you done the following, if applicable
***(optional: add explanation between parentheses)***

* [ ] Added automated tests? *( )*
* [ ] Tested manually on mobile? *( )*
* [x] Checked for performance implications? *( )*
* [ ] Checked for security implications? *( )*
* [ ] Updated the documentation/wiki? *( )*

#### TODOs / next steps:

* [ ] *your TODO here*

AndrewGibson27 commented on static/sass/6-components/_footer.scss:
```diff
@@ -1,6 +1,14 @@
-    margin: $size-xs 0;
+  
+  &__logo {
+    font-size: 4rem;
```
Do you hate the idea of using an inline style to dictate the size of comically large icons like this one? I left a [comment](https://github.com/texastribune/ds-toolbox/pull/25#issuecomment-497519062) about that strategy in this PR -- not sure if it's a good one. Also doesn't matter much.

ashley-hebler replied:
I don't love it, but I understand what you mean. Tis changed.

AndrewGibson27 commented on static/sass/6-components/_footer.scss:
```diff
@@ -1,6 +1,14 @@
+  
+  &__logo {
+    font-size: 4rem;
+    margin-bottom: 2.6rem;
```
Could use a utility class here, I think. I use `has-b-btm-marg` in the portal.

ashley-hebler replied:
good catch!

AndrewGibson27 commented on static/sass/6-components/_footer.scss:
```diff
@@ -1,6 +1,14 @@
+    font-size: 4rem;
+    margin-bottom: 2.6rem;
+  }
+  &__icon {
```
I'm torn here. Part of me thinks social icons in the footer will be common enough that the style for it should live at the DS level. Part of me thinks it'll make more sense at the repo level given, for example, the UMP footer doesn't have social icons. IDK.

ashley-hebler replied:
I'm torn too! That's a tough one. I like the idea of it being in the DS because that'd be the last remaining style, allowing it to only pull from the DS with no repo-specific code. Same case with the /texastribune footer. I don't like UMP getting unused CSS, but it _is_ only 1 declaration 🤔

AndrewGibson27 replied:
Up to you, but my final vote is for adding to DS. It wouldn't shock me at all if someone asks me to add the social icons to the UMP footer, in which case there will be no unused DS code!

AndrewGibson27 commented:
Looks good! I noticed some two-line text gets a little jammed on small screens. That's not something I encountered in UMP because none of the anchor text is long enough. Could either drop some line-height on that `a` rule at the DS level or get crazy and stack those columns like I do on UMP. Or nothing! Or something else!

ashley-hebler commented:
@AndrewGibson27, thanks for taking a look. Yes line-height was not the way to go on the footer links spacing. I added something to remedy that greater issue [here](https://github.com/texastribune/ds-toolbox/pull/29). Hopefully it doesn't gross you out too much.

I can also sneak that `c-site-footer__icon` right spacing into the DS too if you like. See my reply on that and let me know what ya think. Other than those I implemented the other fixes 👍

AndrewGibson27 commented:
This footer is of course still using the old underscore-ridden grid system. I went ahead and torched that in my UMP footer in favor of CSS grid, but I totally understand if that's out of scope for this PR.

ashley-hebler commented:
> This footer is of course still using the old underscore-ridden grid system. I went ahead and torched that in my UMP footer in favor of CSS grid, but I totally understand if that's out of scope for this PR.

Yes, in my mental map (that no one can see 😆) we tackle all the `grid_row`, `col` instances at the same time. And tbh, I've been putting it off because it sounds hard.

I'll add that icon rule to the DS. And then we should be all set for this. Thanks again!
